mod.ts

```diff
@@ -1,61 +1,156 @@
 // deno-lint-ignore-file no-explicit-any
 
-export interface Computation<T = any> {
-  [Symbol.iterator](): Iterator<Control, T, any>;
+interface Thunk<V = unknown> {
+  method: "next" | "throw";
+  iterator: Iterator<Control, unknown, unknown>;
+  value: V | Error;
+  caller?: Thunk;
+}
+
+export interface Computation<T = any, C = Control> {
+  [Symbol.iterator](): Iterator<C, T, any>;
 }
 
 export interface Continuation<T = any, R = any> {
   (value: T): R;
 }
 
+export interface ContinuationTail<T = any, R = any> extends Continuation<T, R> {
+  tail(value: T): void;
+}
+
```

Comment:
Is there a downside to rolling this into the `Continuation` interface?

Reply:
`oneshot` requires a `Continuation` but I can just inline the type in that one case since it's just `(t: T) => R`